### PR TITLE
feat: Assign badges to users

### PR DESCRIPTION
**Pull request by sanderelderman**
Features:
- Admin route to give a badge to a user.
- Route for retrieving all badges of a user.
- Moved application, badges and users service to the data module

### CODE REVIEW

File: libs/gamify/auth/src/lib/auth.module.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
-import { UsersModule } from '@gamify/users';
+// import { UsersModule } from '@gamify/users';
```
Comment (mschilling) [inline libs/gamify/auth/src/lib/auth.module.ts:1]: Can be removed

File: libs/gamify/data/src/lib/services/applications.service.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
+import { forwardRef, Inject, Injectable } from '@nestjs/common';
+import { UsersService } from './users.service';
 
 @Injectable()
 export class ApplicationsService {
 
-    constructor(private readonly data: DataService, private readonly usersService: UsersService) {}
+    constructor(
+        private readonly data: DataService, 
+        @Inject(forwardRef(() => UsersService)) private readonly usersService: UsersService
```
Comment (mschilling) [inline libs/gamify/data/src/lib/services/applications.service.ts:12]: Is this (still) necessary? (forwardRef)

File: libs/gamify/data/src/lib/services/users.service.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    constructor(private readonly data: DataService) {}
+
+    findOne(id: number) {
+        return this.data.user.findUnique({
+            where: { id }
+        });
+    }
+
+    giveBadgeToUser(userId: number, badgeId: number) {
```
Comment (mschilling) [inline libs/gamify/data/src/lib/services/users.service.ts:15]: Perhaps `assignBadgeToUser` ?

File: libs/gamify/data/src/lib/services/applications.service.ts (hunk 2 above)
Comment (sanderelderman) [reply libs/gamify/data/src/lib/services/applications.service.ts]: It is not, removed it in the next commit

File: libs/gamify/data/src/lib/services/users.service.ts (hunk 3 above)
Comment (sanderelderman) [reply libs/gamify/data/src/lib/services/users.service.ts]: Good suggestion!

**Pull request merged**